nats.go

```diff
@@ -3476,3 +3477,17 @@ func (nc *Conn) Barrier(f func()) error {
 	nc.mu.Unlock()
 	return nil
 }
+
+// GetCID returns the client ID assigned by the server the client
+// is currently connected to. Note that the value would change if
+// client reconnects. Also, returned value may be 0 if connecting
+// to a server pre 1.2.0, which did not send the CID back to the
+// client.
+func (nc *Conn) GetCID() (uint64, error) {
+	nc.mu.Lock()
+	defer nc.mu.Unlock()
+	if nc.isClosed() {
```

Comment:
If 0 return an error that it was not provided.

Reply:
Done